Block Bindings: Accept client ID as parameter for `useBlockBindingsUtils`

SantosGuillamot opened the pull request:
## What?
As discussed [here](https://github.com/WordPress/gutenberg/pull/64102#issuecomment-2306631817), I'm adding the possibility of passing a `clientId` to `useBlockBindingsUtils` instead of always relying on the edit context.

Additionally, I added a few unit tests to cover the different use cases.

## Why?
It is a common pattern in other APIs and it is missing in this one.

## How?
I added some logic to rely on the parameter and, if it isn't provided, use the client ID returned by `useBlockEditContext`.

Apart from that, I added unit tests.

## Testing Instructions
It is a bit hard to test this manually because code needs to run inside a component or a hook. I added unit tests to cover the different use cases. Additionally, post meta and pattern overrides are using these utils and are covered with e2e tests.

SantosGuillamot committed:
Pass `clientId` to `useBlockBindingsUtils`

SantosGuillamot committed:
Add unit tests for `useBlockBindingsUtils`

gziolo commented on packages/block-editor/src/utils/test/use-block-bindings-utils.js:
```diff
@@ -0,0 +1,176 @@
+
+	let clientId;
+	beforeEach( async () => {
+		const block = createBlock( 'core/paragraph', {
```
Super nuanced nit. I believe that you can use here `dispatch( blockEditorStore ).resetBlocks( [ block ] );` instead of `insertBlocks`.

This way `dispatch( blockEditorStore ).resetBlocks( [] );` could be run only once in `afterAll` instead of `afterEach`.

SantosGuillamot replied:
That's a good one 🙂 If I understood you correctly, it has been addressed in this commit: https://github.com/WordPress/gutenberg/pull/65818/commits/e52bab20a76c2b6426e691ad70fa1a98d85cf77d

gziolo replied:
Yes, exactly that 👍🏻

gziolo reviewed (approved):
That's a nice addition. I'm not sure why it's still a draft. To me it's good to go.

SantosGuillamot committed:
Use `resetBlocks` instead of `insertBlocks` in unit test

SantosGuillamot commented:
> I'm not sure why it's still a draft. To me it's good to go.

Because I didn't have time to properly review it after pushing and then I forgot to mark it as ready 😄 I have just done that and addressed the comment.
